crates/common/src/signer/mod.rs

```diff
@@ -0,0 +1,122 @@
+use std::{
+    error::Error,
+    fmt::{self, LowerHex},
+};
+
+use eyre::Context;
+use serde::{Deserialize, Serialize};
+use ssz_derive::{Decode, Encode};
+use tree_hash::TreeHash;
+
+pub mod schemes;
+#[allow(clippy::module_inception)]
+mod signer;
+
+pub use schemes::{bls::BlsSecretKey, ecdsa::EcdsaSecretKey};
+pub use signer::{ConsensusSigner, Signer};
+
+pub type Pubkey<T> = <T as SecretKey>::PublicKey;
+
+pub trait SecretKey {
```

Comment:
Try getting rid of this trait and stick to the enums only (abstraction of the signature scheme with this trait deemed not worth it for now where we have only two variants).